packages/request-client.js/src/api/payment-network/erc20/proxy-info-retriever.ts

```diff
@@ -0,0 +1,90 @@
+import { ethers } from 'ethers';
+import * as Types from '../../../types';
+
+// The ERC20 proxy smart contract ABI fragment containing TransferWithReference event
+const erc20proxyContractAbiFragment = [
+  'event TransferWithReference(address tokenAddress,address to,uint256 amount,bytes indexed paymentReference)',
+];
+
+/**
+ * Retrieves a list of transfer events for an address
```

Comment:
It retrieves events from reference and not address, isn't it ?

Reply:
right !